lib/prx_auth/rails/ext/controller.rb

```diff
@@ -26,51 +31,51 @@ def authenticate!
       end
 
       def current_user
-        return if prx_auth_token.nil?
-
-        PrxAuth::Rails::Token.new(prx_auth_token)
+        prx_auth_token
       end
 
-      def lookup_and_register_accounts_names
-        session[PRX_ACCOUNT_NAME_MAPPING_KEY] =
-          lookup_account_names_mapping
+      def sign_in_user(token)
+        session[PRX_JWT_SESSION_KEY] = token
+        accounts_for(current_user.resources)
```

Comment:
On sign-in, attempt to lookup metadata for all the user's account_ids.  But we can still lookup other accounts we're not in later on.